tauri-app/src/lib/components/ModalVaultDepositGeneric.svelte

```diff
@@ -1,69 +1,117 @@
 <script lang="ts">
   import { Button, Modal, Label } from 'flowbite-svelte';
   import InputTokenAmount from '$lib/components/InputTokenAmount.svelte';
-  import { vaultDeposit } from '$lib/services/vault';
+  import { vaultDeposit, vaultDepositApproveCalldata, vaultDepositCalldata } from '$lib/services/vault';
   import InputToken from '$lib/components/InputToken.svelte';
   import InputVaultId from '$lib/components/InputVaultId.svelte';
-  import ButtonLoading from '$lib/components/ButtonLoading.svelte';
+  import { orderbookAddress } from '$lib/stores/settings';
+  import { checkAllowance, ethersExecute } from '$lib/services/ethersTx';
+  import { toasts } from '$lib/stores/toasts';
+  import ModalExecute from './ModalExecute.svelte';
 
   export let open = false;
   let vaultId: bigint = 0n;
   let tokenAddress: string = '';
   let tokenDecimals: number = 0;
   let amount: bigint;
   let isSubmitting = false;
+  let selectWallet = false;
 
   function reset() {
-    vaultId = 0n;
-    tokenAddress = '';
-    tokenDecimals = 0;
-    amount = 0n;
-    isSubmitting = false;
     open = false;
+    if (!isSubmitting) {
+      vaultId = 0n;
+      tokenAddress = '';
+      tokenDecimals = 0;
+      amount = 0n;
+      selectWallet = false;
+    }
   }
 
-  async function execute() {
+  async function executeLedger() {
     isSubmitting = true;
     try {
       await vaultDeposit(vaultId, tokenAddress, amount);
-      reset();
       // eslint-disable-next-line no-empty
     } catch (e) {}
     isSubmitting = false;
+    reset();
   }
-</script>
 
-<Modal title="Deposit to Vault" bind:open outsideclose size="sm" on:close={reset}>
-  <div>
-    <h5 class="mb-2 w-full text-xl font-bold tracking-tight text-gray-900 dark:text-white">
-      Vault ID
-    </h5>
-    <InputVaultId bind:value={vaultId} />
-  </div>
+  async function executeWalletconnect() {
+    isSubmitting = true;
+    try {
+      if (!$orderbookAddress) throw Error("Select an orderbook to deposit");
+      const allowance = await checkAllowance(amount, tokenAddress, $orderbookAddress);
+      if (!allowance) {
+        const approveCalldata = await vaultDepositApproveCalldata(vaultId, tokenAddress, amount) as Uint8Array;
+        const approveTx = await ethersExecute(approveCalldata, tokenAddress);
+        toasts.success("Approve Transaction sent successfully!");
+        await approveTx.wait(1);
+      }
 
-  <div>
-    <h5 class="mb-2 w-full text-xl font-bold tracking-tight text-gray-900 dark:text-white">
-      Token
-    </h5>
-    <InputToken bind:address={tokenAddress} bind:decimals={tokenDecimals} />
-  </div>
+      const depositCalldata = await vaultDepositCalldata(vaultId, tokenAddress, amount) as Uint8Array;
+      const depositTx = await ethersExecute(depositCalldata, $orderbookAddress);
+      toasts.success("Transaction sent successfully!");
```

Comment:
It would be great to use the same styling for transaction status notices for both ledger and wallet connect transactions, but it would require a some work to wrap ethers transactions, determine their progress state, and pass that up to a transaction status notice.

I feel like for now this is good and we can do that in a subsequent PR.

Reply:
#482